relentless/core/variable.py

```diff
@@ -290,9 +290,15 @@ def value(self):
 
     def derivative(self, var):
         if var is self.a:
```

Comment:
I think this logic isn't quite right, since we need to make sure the objects are *exactly* the same, and not just having the same value. This should do the trick:
```py
if var is self.a and var is self.b:
    return 1.0
elif var is self.a or var is self.b:
    return 0.5
else:
    return 0.0
```
Could you also add to the unit tests a case where you have two variables that have the same value (but are different objects) in addition to the case where they are exactly the same object? Both cases will have the same value but different derivatives.